libvmaf/test/test_framesync.c

```diff
@@ -1,90 +1,139 @@
-        .index_offsets = index_offsets,
-        .index_offsets_cnt = index_offsets_cnt,
+static char *test_framesync_create_process_and_destroy()
+{
+    int err, frame_index;
+
+    VmafThreadPool *pool;
+	VmafFrameSyncContext *framesync;
+    unsigned n_threads = 2;
+
+    err = vmaf_thread_pool_create(&pool, n_threads);
+    mu_assert("problem during vmaf_thread_pool_init", !err);
+
+	err = vmaf_framesync_init(&framesync);
+	mu_assert("problem during vmaf_framesync_init", !err);
+	
+	//loop over frames to be tested
+	for(frame_index = 0; frame_index <NUM_TEST_FRAMES; frame_index++)
+	{
+		uint8_t *pic_a = malloc(FRAME_BUF_LEN);
+		uint8_t *pic_b = malloc(FRAME_BUF_LEN);
+
+		printf("Processing frame %d\n\n", frame_index);
+		
+		memset(pic_a, frame_index, FRAME_BUF_LEN);
+		memset(pic_b, (frame_index), FRAME_BUF_LEN);
+		
+		struct ThreadData data = {
+            .ref = pic_a,
+            .dist = pic_b,
+            .index = frame_index,
+            .framesync = framesync,
+            .err = 0,
     };
 
-    err = vmaf_framesync_init(&fs_ctx, cfg);
-    mu_assert("problem during vmaf_framesync_init", !err);
-
-    mu_assert("no callbacks should have been executed yet, "
-              "vmaf_framesync_init",
-              most_recent_callback_index == -1);
-
-    struct my_cookie *my_data = NULL;
-    err = vmaf_framesync_fetch_data(fs_ctx, (void**)&my_data);
-    mu_assert("problem during vmaf_framesync_fetch_data", !err);
-    mu_assert("fetched data pointer is still null", my_data);
-
-    err = vmaf_framesync_submit_data(fs_ctx, my_data, 0);
-    mu_assert("problem during vmaf_framesync_submit_data", !err);
-
-    mu_assert("no callbacks should have been executed yet, "
-              "callback dependencies unmet",
-              most_recent_callback_index == -1);
-
-    err = vmaf_framesync_fetch_data(fs_ctx, (void**)&my_data);
-    mu_assert("problem during vmaf_framesync_fetch_data", !err);
-    mu_assert("fetched data pointer is still null", my_data);
-
-    err = vmaf_framesync_submit_data(fs_ctx, my_data, 1);
-    mu_assert("problem during vmaf_framesync_submit_data", !err);
-
-    mu_assert("no callbacks should have been executed yet, "
-              "callback dependencies unmet",
-              most_recent_callback_index == -1);
-
-    err = vmaf_framesync_fetch_data(fs_ctx, (void**)&my_data);
-    mu_assert("problem during vmaf_framesync_fetch_data", !err);
-    mu_assert("fetched data pointer is still null", my_data);
-
-    err = vmaf_framesync_submit_data(fs_ctx, my_data, 2);
-    mu_assert("problem during vmaf_framesync_submit_data", !err);
-
-    mu_assert("callback should have been executed for index == 1, "
-              "callback dependencies are met: (0, 1, 2)",
-              most_recent_callback_index == 1);
-
-    err = vmaf_framesync_close(fs_ctx);
-    mu_assert("problem during vmaf_framesync_close", !err);
+		err = vmaf_thread_pool_enqueue(pool, framesync_proc_func, &data, sizeof(ThreadData));
```

Comment:
Note: There is inconsistent use of spaces and tabs in a few places

Reply:
Tabs have been converted to spaces.